Services/Badge/classes/class.ilBadgeHandler.php

```diff
@@ -541,14 +545,14 @@ public function sendNotification(
                     $url = new ilNotificationLink(new ilNotificationParameter('badge_notification_badges_goto', [], 'badge'), $url);
                     $osd_params = ["badge_list" => implode(", ", $user_badges)];
 
-                    $notification = new ilNotificationConfig(BadgeNotificationProvider::NOTIFICATION_TYPE);
+                    $notification = $provider->getNotificationConfig();
                     $notification->setTitleVar("badge_notification_subject", [], "badge");
                     $notification->setShortDescriptionVar("badge_notification_osd", $osd_params, "badge");
                     $notification->setLongDescriptionVar("");
                     $notification->setLinks([$url]);
                     $notification->setIconPath(ilUtil::getImagePath('icon_bdga.svg'));
-                    $notification->setValidForSeconds(ilNotificationConfig::TTL_SHORT);
-                    $notification->setVisibleForSeconds(ilNotificationConfig::DEFAULT_TTS);
+                    $notification->setProviderKey(self::BADGES);
```

Comment:
IMO this does not help. The notification emitted through `Services/Notifications` here must be provided with a **unique** `Identification`. Instead, you pass a static `badges` as key. How should the `Services/Badges` in turn emit a signal to `Services/Notifications`, that the notification sent in the past is not longer valid and should be removed?